apps/frontend/app/(website)/studio/src/components/CodemodButton/RepositoryModal.tsx

```diff
@@ -0,0 +1,280 @@
+import Modal from "@studio/components/Modal";
+import { Button } from "@studio/components/ui/button";
+import type { GithubRepository } from "be-types";
+import { matchSorter } from "match-sorter";
+
+import Input from "@/components/shared/Input";
+import {
+  Combobox,
+  ComboboxItem,
+  ComboboxList,
+  ComboboxProvider,
+} from "@ariakit/react";
+import * as RadixSelect from "@radix-ui/react-select";
+import { CheckIcon, ChevronUpDownIcon, SearchIcon } from "@studio/icons";
+import type { ToVoid } from "@studio/types/transformations";
+import { startTransition, useMemo, useState } from "react";
+
+export type RepositoryModalProps = {
+  hideRepositoryModal: VoidFunction;
+  isRepositoryModalShown: boolean;
+  repositoriesToShow: GithubRepository[];
+  selectRepository: ToVoid<GithubRepository["full_name"]>;
+  selectedRepository?: GithubRepository;
+  branchesToShow: string[];
+  selectBranch: ToVoid<string>;
+  selectedBranch?: string;
+  codemodNameInput: string;
+  setCodemodNameInput: ToVoid<string>;
+  targetPathInput: string;
+  setTargetPathInput: ToVoid<string>;
+  onRunCodemod: VoidFunction;
+};
+export const RepositoryModal = ({
+  hideRepositoryModal,
+  isRepositoryModalShown,
+  repositoriesToShow,
+  selectRepository,
+  selectedRepository,
+  branchesToShow,
+  selectBranch,
+  selectedBranch,
+  // targetPathInput,
+  // setTargetPathInput,
+  codemodNameInput,
+  setCodemodNameInput,
+  onRunCodemod,
+}: RepositoryModalProps) => {
+  const [repoValueToFilterBy, setRepoValueToFilterBy] =
+    useState<GithubRepository["full_name"]>();
+  const [branchValueToFilterBy, setBranchValueToFilterBy] = useState<string>();
+  const [repoSelectorOpen, setRepoSelectorOpen] = useState(false);
+  const [branchSelectorOpen, setBranchSelectorOpen] = useState(false);
+
+  const repoMatches = useMemo(() => {
```

Comment:
isnt this memoization breaking the search feature? 
This kind of operations are super fast in js, I don't think we need to memo it